### PR TITLE
add download, copy image and meta html buttons

### LINKED ISSUE #4: Download, URL, and HTML buttons
> I was thinking if we should add a download option there as well. Let me know if it's required or will > be of some help then I will be happy to work on it as well.

_Originally posted by @rajat2502 in https://github.com/creativecommons/og-image-generator/pull/3#issuecomment-705216732_

---

@rajat2502 I agree this is a good idea. The _best practice_ for using this is to not download, but to instead use the url in the meta tag, like so:

```
<meta
  property="og:image"
  content="https://cc-og-image.vercel.app/Introducing%20**New**%20Feature.png?theme=light&md=1&fontSize=100px&images=https%3A%2F%2Fcc-vocabulary.netlify.app%2Flogos%2Fcc%2Flogomark.svg%23logomark"
/>
```

Right now, clicking on the image preview will copy the url to the user's clipboard. This is good, but it's hard to notice. I propose we add **three** buttons, that do the following:

- Download Image
- Copy URL
- Copy Meta Tag (provides the full html for `og:image`

Each of these buttons should show a toast/pop-up notification, like clicking on the image currently does.

### PR DESCRIPTION
**Pull request by rajat2502**
## Fixes

Fixes #4  by @zackkrida 

## Description

Added download, copy image URL and meta tags buttons to the project.

## Technical details

Added action buttons for download, copying the image, and copying meta tags.

## Screenshots

image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by rajat2502** [comment]
Hey @zackkrida, please review and let me know if I need to make any changes.

File: web/index.ts [hunk 1]
```diff
@@ -254,6 +276,36 @@ const App = (_: any, state: AppState, setState: SetState) => {
...
+      setTimeout(() => setState({ showToast: false }), 3000)
+    } else {
+      window.open(url.href, '_blank')
+    }
+    return false
+  }
+
+  const copyMetaTag = (e: Event) => {
+    e.preventDefault()
+    const html = `<meta property="og:image" content=${url.href}/>`
```
Comment (zackkrida) [inline web/index.ts:297]:
```suggestion
    const html = `<meta property="og:image" content=${url.href}/>`
    const html = `<meta property="og:image" content="${url.href}" />`
```

**Review by zackkrida — APPROVED** [review]
Looks great, if you can just fix the one small suggestion

**Comment by rajat2502** [comment]
Thanks, @zackkrida.

**Pull request merged**